experiment/migrator/migrator.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python2
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Migrates information from Testgrid's Config.yaml to a subdirectory of Prow Jobs
+
+Only moves Dashboard Tabs
+Skips any Dashboard Tab Configuration that contains unusual keys, even if they're incorrect keys.
+"""
+
+import re
+from os import walk
+import ruamel.yaml
+
+PROW_DIR = "../../config/jobs"
+TESTGRID_CONFIG = "../../testgrid/config.yaml"
+
+def main():
+    confFile = open(TESTGRID_CONFIG, "r+")
+    config = ruamel.yaml.load(confFile, Loader=ruamel.yaml.RoundTripLoader, preserve_quotes=True)
+
+    for dashboard in config["dashboards"]:
+        dashNameA = dashboard["name"]
+        if "dashboard_tab" in dashboard:
+            for dashTab in dashboard["dashboard_tab"][:]:
+                if assertTabKeys(dashTab):
+                    moveTab(dashboard, dashTab, dashNameA)
+
+    confFile.seek(0)
+    ruamel.yaml.dump(config, confFile, Dumper=ruamel.yaml.RoundTripDumper)
+    confFile.truncate()
+    confFile.close()
+
+
+def moveTab(dashboard, dashTab, dashboardName):
+    """ moveTab determines if the dashboard tab matches a prow job in a file.
+    If it does, it patches the prow job to include the necessary annotations
+    It also deletes from the config the dashboard tab, so there are no duplicates.
+
+    It has a hardcoded exception for fejta-bot-periodics, because bugs
+    """
+    prowJobName = dashTab["test_group_name"]
+    prowJobFileName = findProwJob(prowJobName, PROW_DIR)
+    # Ruamel won't be able to successfully dump fejta-bot-periodics
+    # See https://bitbucket.org/ruamel/yaml/issues/258/applying-json-patch-breaks-comment
+    if (prowJobFileName != ""
+            and prowJobFileName != (PROW_DIR + "/kubernetes/test-infra/fejta-bot-periodics.yaml")):
```

Comment:
consider returning on the not of this, such that we can avoid nesting everything else below under this if